### PR TITLE
fix monk

### PR DESCRIPTION
**Pull request by qndel**
Fixes https://github.com/diasurgical/devilutionX/issues/1658

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -31,15 +31,31 @@ bool deathflag;
...
 /** Maps from player class to letter used in graphic files. */
 const char CharChar[] = {
-	'W',
-	'R',
-	'S',
-	'M',
+	'W', // warrior
+	'R', // rogue
+	'S', // sorcerer
+	'M', // monk
```
Comment (julealgon) [inline Source/player.cpp:58]: Would be nice to have these descriptions be part of a different PR.

File: Source/player.cpp [hunk 2]
```diff
@@ -444,12 +460,13 @@ static DWORD GetPlrGFXSize(HeroClass c, const char *szCel)
...
 		if (gbIsSpawn && a != &ArmourChar[0])
 			break;
-		for (w = &WepChar[0]; *w; w++) { // BUGFIX loads non-existing animagions; DT is only for N, BT is only for U, D & H (fixed)
+		for (w = &WepChar[0]; *w; w++) { // BUGFIX loads non-existing animagions; DT is only for N, BL is only for U, D & H (fixed)
 			if (szCel[0] == 'D' && szCel[1] == 'T' && *w != 'N') {
 				continue; //Death has no weapon
 			}
 			if (szCel[0] == 'B' && szCel[1] == 'L' && (*w != 'U' && *w != 'D' && *w != 'H')) {
-				continue; //No block without weapon
+				if (c != HeroClass::Monk || (*w != 'T' && *w != 'N')) // monk can block with a staff and unarmed
```
Comment (julealgon) [inline Source/player.cpp:468]: What if the monk has a single one-handed weapon equiped?

Comment (qndel) [reply Source/player.cpp:468]: then he can't block?

Comment (julealgon) [reply Source/player.cpp:468]: > A Monk can also block with a staff and with at least one hand bare.

http://www.bigd-online.com/JG/JGFrame.html

Section 2.2.1

Comment (qndel) [reply Source/player.cpp:468]: jesus, so when he CAN'T block? lol

File: Source/player.cpp [hunk 3]
```diff
@@ -441,15 +458,27 @@ static DWORD GetPlrGFXSize(HeroClass c, const char *szCel)
...
+		canBlock.insert('M');
+		canBlock.insert('N');
+		canBlock.insert('T');
+	}
+
 	for (a = &ArmourChar[0]; *a; a++) {
 		if (gbIsSpawn && a != &ArmourChar[0])
 			break;
-		for (w = &WepChar[0]; *w; w++) { // BUGFIX loads non-existing animagions; DT is only for N, BT is only for U, D & H (fixed)
+		for (w = &WepChar[0]; *w; w++) { // BUGFIX loads non-existing animagions; DT is only for N, BL is only for U, D & H (fixed) - for monk BL is for all except B/A
```
Comment (AJenbo) [inline Source/player.cpp:476]:
```suggestion
		for (w = &WepChar[0]; *w; w++) {
```

File: Source/player.cpp [hunk 4]
```diff
@@ -441,15 +458,27 @@ static DWORD GetPlrGFXSize(HeroClass c, const char *szCel)
...
 			break;
-		for (w = &WepChar[0]; *w; w++) { // BUGFIX loads non-existing animagions; DT is only for N, BT is only for U, D & H (fixed)
+		for (w = &WepChar[0]; *w; w++) {
 			if (szCel[0] == 'D' && szCel[1] == 'T' && *w != 'N') {
 				continue; //Death has no weapon
 			}
-			if (szCel[0] == 'B' && szCel[1] == 'L' && (*w != 'U' && *w != 'D' && *w != 'H')) {
-				continue; //No block without weapon
+			if (szCel[0] == 'B' && szCel[1] == 'L' && canBlock.find(*w) == canBlock.end()) {
+					continue; //No block
```
Comment (AJenbo) [inline Source/player.cpp:481]:
```suggestion
				continue; // No block animation
```

**Comment by AJenbo** [comment]
Nice cleanup, much easier to understand the code now. Your commits and commit messages are shit though :D

**Review by AJenbo — APPROVED** [review]

**Comment by qndel** [comment]
> Nice cleanup, much easier to understand the code now. Your commits and commit messages are shit though :D

well, it gets squashed anyway right? ;d

**Comment by AJenbo** [comment]
Yeah, but not even the first commit has a sensible message :D
